### PR TITLE
Remove unused hints

### PR DESCRIPTION
**Pull request by KoBeWi**

### CODE REVIEW

**Comment by akien-mga** [comment]
I think that the enum values should be made continuous again, we can break compat in 4.0.

File: core/object/object.h [hunk 1]
```diff
@@ -127,9 +123,9 @@ enum PropertyUsageFlags {
 	PROPERTY_USAGE_READ_ONLY = 1 << 29, // Mark a property as read-only in the inspector.
 	PROPERTY_USAGE_ARRAY = 1 << 30, // Used in the inspector to group properties as elements of an array.
 
-	PROPERTY_USAGE_DEFAULT = PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_NETWORK,
-	PROPERTY_USAGE_DEFAULT_INTL = PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_NETWORK | PROPERTY_USAGE_INTERNATIONALIZED,
-	PROPERTY_USAGE_NO_EDITOR = PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_NETWORK,
+	PROPERTY_USAGE_DEFAULT = PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR,
+	PROPERTY_USAGE_DEFAULT_INTL = PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_INTERNATIONALIZED,
+	PROPERTY_USAGE_NO_EDITOR = PROPERTY_USAGE_STORAGE,
```
Comment (KoBeWi) [inline core/object/object.h:128]: I just realized `PROPERTY_USAGE_NO_EDITOR` is now basically an alias to `PROPERTY_USAGE_STORAGE`. Should it be removed maybe?

File: core/extension/gdnative_interface.h [hunk 2]
```diff
@@ -252,11 +252,8 @@ typedef void *GDNativeExtensionClassLibraryPtr;
 typedef enum {
 	GDNATIVE_EXTENSION_METHOD_FLAG_NORMAL = 1,
 	GDNATIVE_EXTENSION_METHOD_FLAG_EDITOR = 2,
-	GDNATIVE_EXTENSION_METHOD_FLAG_NOSCRIPT = 4,
 	GDNATIVE_EXTENSION_METHOD_FLAG_CONST = 8,
-	GDNATIVE_EXTENSION_METHOD_FLAG_REVERSE = 16, /* used for events */
 	GDNATIVE_EXTENSION_METHOD_FLAG_VIRTUAL = 32,
-	GDNATIVE_EXTENSION_METHOD_FLAG_FROM_SCRIPT = 64,
```
Comment (akien-mga) [inline core/extension/gdnative_interface.h:259]: We don't need to keep holes, this is new in 4.0.

Comment (akien-mga) [reply core/extension/gdnative_interface.h:259]: This is also now out of sync with Object.

File: core/object/object.h [hunk 3]
```diff
@@ -101,8 +99,6 @@ enum PropertyUsageFlags {
 	PROPERTY_USAGE_NONE = 0,
 	PROPERTY_USAGE_STORAGE = 1,
 	PROPERTY_USAGE_EDITOR = 2,
-	PROPERTY_USAGE_NETWORK = 4,
-	PROPERTY_USAGE_EDITOR_HELPER = 8,
```
[the remark this answers is not in the document]
Comment (KoBeWi) [reply core/object/object.h:105]: Yeah that was the intention, I just missed this.

**Comment by YuriSizov** [comment]
Maybe we can reorder some of them to be presented in a more organized manner? Since this is a clean-up PR anyway...

**Comment by KoBeWi** [comment]
If you have ideas for better order I'm open to suggestions.

**Comment by akien-mga** [comment]
Yeah I've been meaning to get this thoroughly cleaned up/reordered/renamed but I can't muster the brain capacity to do a proposal for this.

Related: #30203.

**Comment by KoBeWi** [comment]
Maybe we could merge this now and debate about ordering later? Rebasing is troublesome >_>

**Review by akien-mga — APPROVED** [review]
Let's merge and do further cleanup later.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**